snowfall/training/diagnostics.py

```diff
@@ -0,0 +1,77 @@
+from typing import Dict
+
+import torch
+from torch import nn
+
+
+def measure_weight_norms(model: nn.Module, norm: str = 'l2') -> Dict[str, float]:
+    """
+    Compute the norms of the model's parameters.
+    Norms where it's applicable are normalized by the number
```

Comment:
Not sure if it's best to normalize by number of weights.  IMO it might be more meaningful to display without normalization.